rpc_helper/patch.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2022 Camptocamp SA
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import odoo
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+
+def protected__execute_cr(cr, uid, obj, method, *args, **kw):
+    # Same as original func in odoo.service.model.execute_cr
+    odoo.api.Environment.reset()  # clean cache etc if we retry the same transaction
+    recs = odoo.api.Environment(cr, uid, {}).get(obj)
+    if recs is None:
+        raise UserError(_(f"Object {obj} doesn't exist"))
+    # custom code starts here
+    if not _rpc_allowed(recs, method):
+        raise UserError(_(f"RPC call on {obj} is not allowed"))
```

Comment:
Wrong interpolation of the string for a translation. You should rollback and simply move the param out of the parenthesis
```suggestion
        raise UserError(_("RPC call on %s is not allowed") % obj)
```

Reply:
sure, just adjusted 👍🏿